### ISSUE TITLE
Segfault when blitting surface of size (3712, 3712) to another surface at position (2.43484e+09, 2.45954e+09).

### ISSUE THREAD

**Issue by NOTMEE12** [pygame-community/pygame-ce#2288]
**Environment:**

You can get some of this info from the text that pops up in the console when you run a pygame program.

- **Operating system** (e.g. Windows, Linux(Debian), Linux(Ubuntu), Mac): Windows 11
- **Python version** (e.g. 3.7.9, 3.8.5) : 3.11.3
- **SDL version** (e.g. SDL 2.0.12): 2.26.4
- **PyGame version** (e.g. 2.0.0.dev10, 1.9.6): 2.3.0
- **Relevant hardware** (e.g. if reporting a bug about a controller, tell us the brand & name of it):

**Current behavior:**

When blitting surface of size `(3712, 3712)` to a position `(2.43484e+09, 2.45954e+09)` it returns a segfault.

**Expected behavior:**

It shouldn't return a segfault, I guess?

**Steps to reproduce:**

1. Create a surface of size `(3712, 3712)`
2. Blit it onto a surface on `(2.43484e+09, 2.45954e+09)`

It will only segfault when you blit on a surface that is smaller than half of the 3712 (eg. `(3712, 3712/2)`).
It may also segfault on other sizes but I only tested it on these.

**Test code**

If possible add a simple test program that shows the problem described in this report.
```py
import pygame

pygame.init()

screen = pygame.Surface((500, 500))

surf = pygame.Surface((3712, 3712))

screen.blit(surf, (0, 0))

screen.blit(surf, (2.43484e+09, 2.45954e+09))
```

**Stack trace/error output/other error logs**

```
pygame-ce 2.3.0 (SDL 2.26.4, Python 3.11.3)
Fatal Python error: pygame_parachute: (pygame parachute) Segmentation Fault
Python runtime state: initialized

Current thread 0x00000998 (most recent call first):
  File "D:\WorldD\test.py", line 11 in <module>

Extension modules: pygame.base, pygame.constants, pygame.rect, pygame.rwobject, pygame.surflock, pygame.bufferproxy, pygame.math, pygame.surface, pygame.display, pygame.draw, pygame.joystick, pygame.event, pygame.imageext, pygame.image, pygame.key, pygame.mouse, pygame.time, pygame.mask, pygame.pixelcopy, pygame.transform, pygame.font, pygame.mixer_music, pygame.mixer, pygame.scrap, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, pygame.system, pygame._freetype (total: 39)

Process finished with exit code -1073740791 (0xC0000409)

```

**Comment by oddbookworm** [comment]
This is likely an issue with SDL itself, a similar segfault was reported before and I opened an issue on the SDL GitHub.
https://github.com/libsdl-org/SDL/issues/7614

**Comment by 1bsyl** [comment]
yes, 
please give a try with the PR: https://github.com/libsdl-org/SDL/pull/7808

**Comment by oddbookworm** [comment]
@1bsyl Unfortunately, that may end up having to wait a little while on our end. Last I saw, we weren't ready for SDL3 yet (unless @Starbuck5, @MyreMylar, or @ankith26 happens to have an SDL3-ready branch that they can test with)

**Comment by oddbookworm** [comment]
Yeah, I tried. Unless versions of the other libs precompiled can be provided, I'm done for the night. I can try again tomorrow night, but no promises

**Comment by Starbuck5** [comment]
I do have an SDL3 branch, but it's out of date by 3 months both our changes and SDL3 changes.

We don't need pygame-ce integration to test this, this is testable just with a C program. The hard part would be building SDL from source to test it, but SDL isn't too difficult to build.

**Comment by ankith26** [comment]
This issue is no longer happening on wheels made by CI on `main`. It's fixed by upgrading to SDL `2.30.x`.
